pgdog/src/backend/pool/replicas/ban.rs

```diff
@@ -0,0 +1,396 @@
+use super::*;
+use parking_lot::RwLock;
+use std::time::Instant;
+
+use tracing::{error, warn};
+
+/// Load balancer target ban.
+#[derive(Clone, Debug)]
+pub struct Ban {
+    inner: Arc<RwLock<BanInner>>,
+    pool: Pool,
+}
+
+impl Ban {
+    /// Create new ban handler.
+    pub(super) fn new(pool: &Pool) -> Self {
+        Self {
+            inner: Arc::new(RwLock::new(BanInner { ban: None })),
+            pool: pool.clone(),
```

Comment:
(For my edification, not review:) Pools are immutable when cloned here, right? so we don't have any issue with the clone diverging from parent?